frontend/src/metabase-types/api/entity-id.ts

```diff
@@ -0,0 +1,24 @@
+const NANOID_ALPHABET =
+  "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const NANOID_LENGTH = 21;
+
+declare const __brand: unique symbol;
+type Brand<T, B> = T & { readonly [__brand]: B };
+type NanoID = Brand<string, "NanoID">;
```

Comment:
2 questions:
1. How does this type work?
    I tried `const: NanoID = "abcdefghijklmnopqrstu"` and this gives me error. I'd expect this to work given it has a type.
2. We export `NanoID` as `BaseEntityId` while the function is named `isNanoID` I think we should use the same name either a `Nano ID` or an `Entity ID`

Reply:
1. The idea is that you can't just create a NanoID - those should really be always coming from the backend. In your case, you've created a `string` that you're trying to specify as a `NanoID`. To make this work, you'll need to explicitly cast the string to a NanoID with `as NanoID/BaseEntityId` or use the `asNanoId` utility below to keep type safety.
2. As I was typing that first answer I see what you mean. I'll stick with `EntityId` for uses outside this file because it's super annoying otherwise

Reply:
Re 1) that makes sense. This ID isn't supposed to be manually created.

Reply:
I'll add a comment at the top of this file just to clarify for anybody else who might have the same idea. It's a very valid way of thinking